hwcontest.md

```diff
@@ -10,59 +10,78 @@ active_tab: homework
 # Compiler Contest 
 
 <p class="text-muted">Start on {{ site.hwdates[5].startdate }}</p>
-<p class="text-muted">Due on {{ site.hwdates[5].deadline }}</p>
-<p class="text-muted">With grace days due on {{ site.hwdates[5].gracedays }}</p>
+<p class="text-muted">Part 1 due on {{ site.hwdates[5].deadline }}</p>
+<p class="text-muted">With grace days, Part 1 due on {{ site.hwdates[5].gracedays }}</p>
+<p class="text-muted">Part 2 due on {{ site.hwdates[6].deadline }}</p>
 
 ## Your Task
 
-1. Create at least 10 new testcases and at most 50 testcases that are distinct from the provided testcases for the homeworks (hw1-hw4). They must pass validation (see below).
-1. Your testcases must pass and are expected to produce valid LLVM output that when compiled to binary produce some output after execution of the binary. 
-1. Testcases expected to fail cannot be submitted to the contest. (_this rule might be dropped_)
-1. The submission file for the Compiler Contest will be uploaded to Coursys.
+1. **Part 1.** Create at least 10 new testcases and at most 20 testcases that are distinct from the provided testcases for the homeworks (hw1-hw4). They must pass validation (see below).
+  1. Your testcases should be valid Decaf code which is not expected to crash. Each one should produce some output when it is executed.
+  1. Your grade for part 1 depends on how many of your testcases pass validation.
+1. **Part 2.** After the testcases have been validated, you will receive a zipfile containing all of the submitted testcases which passed validation. You must run your compiler on these cases and submit the output to Coursys.
+  1. Your grade for part 2 depends on how many of the testcases your compiler passes, and how difficult your testcases are for other students' compilers.
+
+Please note:
+1. Grace days are only available for Part 1. There is a hard deadline on December 10 for the submission of Part 2.
+1. Both parts will be submitted to Coursys. 
+1. Testcases expected to fail cannot be submitted to the contest.
 1. No fuzzing allowed. The testcases are expected to be hand checked or sanitized even if you used an automated method to generate them.
 
 ## The Contest
 
-### Testcase format
+### Part 1: Testcase format
 
 This section specifies the testcase directory structure.
 
 1. The testcase format must be identical to the format for the homeworks, particularly [hw3](hw3.html) and [hw4](hw4.html). 
-1. In your answer directory create two new directories: `testcases/your-name` and `references/your-name` where `your-name` is the name you have been using for the leaderboard.
+1. In your answer directory create two new directories: `testcases/your-name` and `references/your-name` where `your-name` is your SFU username.
 1. The directory `testcases/your-name` should contain the Decaf source files named with a `.decaf` file name suffix.
```

Comment:
...as well as `.in` files containing the inputs for testcases which expect user input. (?)

Reply:
yes. they can include the `.in` files as well for those cases requiring input from stdin.